### PR TITLE
Improve planned enhancements in DG

### LINKED ISSUE #291: [PE-D][Tester A] Hard to see what weddings and persons to select
For some commands like the adding of tags, a person and a wedding needs to be specified. However, as the person list and the wedding lists can only be displayed one at a time, the user needs to remember the name of either the person or the wedding before they can issue the `add` command with the right parameters.

If I wanted to assign a person "jeremy lim" to "Connor & Aishah" wedding, I would need to remember the full name of jeremy to issue the right command. choosing to change the display via "lw" or "list" wouldn't help much as it would mean erasing the current half-typed `add` command to issue a new `list` or `lw` program.

This would slow the process down for the user slightly if they were not familiar with the entries in KnottyPlanner.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: C5hives/ped#2

### LINKED ISSUE #326: [PE-D][Tester D] Unable to close help window without moving the mouse
As the app is optimized for CLI users, would it be better if there is a way to close the help window without moving the mouse?

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: hooyushing/ped#13

### LINKED ISSUE #334: [PE-D][Tester D] Double confirmation of deleting slows down for CLI users, who want it to be fast
I would think that it would slow down the user in some way, maybe there could be a del force n/NAME so it would be faster for the CLI user?

In terminal/linux, there is a force command which allows command to be executed without confirmation

Expected: Fast typing single command allows deletion
Actual: Slowed down by the double confirmation

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: hooyushing/ped#15

### LINKED ISSUE #346: [PE-D][Tester D] Filter command does not allow searches by other fields
If I were a wedding planner and I only have the phone number of the individual, but not the full name, I am unable to search of the person. Maybe it can be improved to allow this feature? :D

Expected output: I am able to search for the person
Actual output: Error message given.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: hooyushing/ped#19

### LINKED ISSUE #349: [PE-D][Tester A] Selecting Persons and Weddings is slow if their full name needs to be typed out
Selecting any item in the application requires the user to know the item's full name, and the commands require the full name to be typed out.

To delete "Jeremy Ng" from the list, the `del` command will need to be `del n/Jeremy Ng". This can become a big issue if the names are too long, and the user needs to type the whole name out. This applies for wedding names as well.

Ultimately, this makes the user perform tasks at a slower rate because of the effort needed to type out many long commands.

It might be helpful to allow the user to select items based on their index in their lists instead of needing the full name.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: C5hives/ped#11

### LINKED ISSUE #351: [PE-D][Tester B] Long input for address become not visible
Very long input for address become non visible

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: JoanneHing/ped#9

### LINKED ISSUE #354: [PE-D][Tester A] UC01 should not loop back to previous steps if selected item is invalid
For the MSS, at extension 3a, the use case should not resume at step 2. Terminating the use case instead should be the better option, else the user will need to select a valid contact to delete in order to end the use case.

Removing the loop will also help to simplify this use case to be the simplest scenario.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: C5hives/ped#12

### LINKED ISSUE #363: [PE-D][Tester D] Unable to copy information from person card
As a wedding planner, I would want fast access to information on the person card, to be able to copy information from the person card to paste elsewhere. But I am unable to do so, and only able to click on it.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: hooyushing/ped#22

### PR DESCRIPTION
**Pull request by siriousguy**
Fixes #291 
Fixes #326  
Fixes #334 
Fixes #346 
Fixes #349 
Fixes #351 
Fixes #354  
Fixes #363

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -220,14 +220,60 @@ Here is an example usage scenario and how `StaticContext` is used in KnottyPlann
...
+1. Prefix Enhancements
 
 Currently: The view wedding command in KnottyPlanners does not require a need for a prefix to represent the wedding. Thus, when associating a wedding to a `Person`, it is done by _tagging_ them to it, making the prefix 't/'. 
 
 Plan: To make the association of `Wedding` and `Person` more intuitive, we will change the prefix from 't/' to 'w/', representing the `Wedding` that a `Person` is associated with.
 
-// Support for Multi-Languages
- 
+2. Forced Deletion and Clearing
+
```
Comment (jonushzw) [inline docs/DeveloperGuide.md:230]: where is point 3

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -220,14 +220,60 @@ Here is an example usage scenario and how `StaticContext` is used in KnottyPlann
...
+**Currently:** As KnottyPlanners is optimised for keyboard usage, interactions with the GUI through direct copying of information is not supported.
+
+**Plan:** To allow the users of KnottyPlanners to copy information directly from the contact or wedding cards, in order to facilitate mouse usage.
+
+**5. Support for Long Inputs**
+
+**Currently:** KnottyPlanners only support inputs up till a reasonable amount of characters. As such, extreme inputs that exceed our limit can potentially hinder the viewing of information when the user does not resize the window.
+
+**Plan:** To handle extreme inputs of all fields with a long character count, ensuring that the information can still remain visible.
```
Comment (ywllowsensor) [inline docs/DeveloperGuide.md:251]: seems abit vague, was this intentional haha
if not can prob say smth like the ui will show the input completely even if its very long

**Review by ywllowsensor — COMMENTED** [review]
JUST ONE NIT

**Comment by jonushzw** [comment]
LGTM!

**Pull request merged**